sentry-android-replay/src/main/java/io/sentry/android/replay/ReplayCache.kt

```diff
@@ -0,0 +1,167 @@
+package io.sentry.android.replay
+
+import android.graphics.Bitmap
+import android.graphics.Bitmap.CompressFormat.JPEG
+import android.graphics.BitmapFactory
+import io.sentry.SentryLevel.DEBUG
+import io.sentry.SentryLevel.ERROR
+import io.sentry.SentryLevel.WARNING
+import io.sentry.SentryOptions
+import io.sentry.android.replay.video.MuxerConfig
+import io.sentry.android.replay.video.SimpleVideoEncoder
+import io.sentry.protocol.SentryId
+import java.io.Closeable
+import java.io.File
+
+internal class ReplayCache(
+    private val options: SentryOptions,
+    private val replayId: SentryId,
+    private val recorderConfig: ScreenshotRecorderConfig,
+    private val encoderCreator: (File) -> SimpleVideoEncoder = { videoFile ->
+        SimpleVideoEncoder(
+            options,
+            MuxerConfig(
+                file = videoFile,
+                recorderConfig = recorderConfig,
+                frameRate = recorderConfig.frameRate.toFloat(),
+                bitrate = 20 * 1000
+            )
+        ).also { it.start() }
+    }
+) : Closeable {
+
+    private val encoderLock = Any()
+    internal var encoder: SimpleVideoEncoder? = null
+
+    internal val replayCacheDir: File? by lazy {
+        if (options.cacheDirPath.isNullOrEmpty()) {
+            options.logger.log(
+                WARNING,
+                "SentryOptions.cacheDirPath is not set, session replay is no-op"
+            )
+            null
+        } else {
+            File(options.cacheDirPath!!, "replay_$replayId").also { it.mkdirs() }
+        }
+    }
+
+    // TODO: maybe account for multi-threaded access
+    internal val frames = mutableListOf<ReplayFrame>()
+
+    fun addFrame(bitmap: Bitmap, frameTimestamp: Long) {
+        if (replayCacheDir == null) {
+            return
+        }
+
+        val screenshot = File(replayCacheDir, "$frameTimestamp.jpg").also {
+            it.createNewFile()
+        }
+        screenshot.outputStream().use {
+            bitmap.compress(JPEG, 80, it)
+            it.flush()
+        }
+
+        val frame = ReplayFrame(screenshot, frameTimestamp)
+        frames += frame
+    }
+
```

Comment:
In Flutter, I think I could use an `addFrame()` method that accepts a file path as an argument.